src/Evolution/Systems/RadiationTransport/M1Grey/Fluxes.cpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/RadiationTransport/M1Grey/Fluxes.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"  // IWYU pragma: keep
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace RadiationTransport {
+namespace M1Grey {
+
+namespace detail {
+void compute_fluxes_impl(
+    gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*> tilde_e_flux,
+    gsl::not_null<tnsr::Ij<DataVector, 3, Frame::Inertial>*> tilde_s_flux,
+    const Scalar<DataVector>& tilde_e,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& tilde_s,
+    const tnsr::II<DataVector, 3, Frame::Inertial>& tilde_p,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
+    const tnsr::II<DataVector, 3, Frame::Inertial>&
+        inv_spatial_metric) noexcept {
+  Variables<tmpl::list<Tags::TildeSVector<Frame::Inertial>>> temp_tensors(
```

Comment:
Could you allocate this once in the wrapper and pass it in as a `gsl::not_null`? That will reduce the number of memory allocations done.

Reply:
So, this is causing me some problems. 
(1) If I do this and implement the function in the hpp file, I'd have to add a lot of header files to the hpp, which I think we are trying to avoid. 
(2) If i try to define it in the cpp file, I run into trouble because the struct ComputeFluxes is templated and the executables using ComputeFluxes fail if the function implementation is in the cpp file and does not explicitly implement the function with the desired NeutrinoSpecies.

I can do case (1) if we are ok with a bloated hpp file. If we want case (2), I'll need some help on how to go around the above problem. [or we have option (3) - keep things as they are]